packages/database/realtime/database.service.ts

```diff
@@ -0,0 +1,312 @@
+import {Injectable} from "@nestjs/common";
+import {
+  ChangeStream,
+  DatabaseService,
+  Document,
+  FilterQuery,
+  ObjectId
+} from "@spica-server/database";
+import {Observable, Subject, Subscription} from "rxjs";
+import {bufferTime, filter, share} from "rxjs/operators";
+import {levenshtein} from "./levenshtein";
+import {late} from "./operators";
+import {ChunkKind, StreamChunk} from "./stream";
+
+/**
+ * Converts the filter to not form
+ * @example { 'subfilter': true } -> { 'fullDocument.subfilter': { '$ne': true } }
+ * @example { 'fullDocument.stars': { '$gt': 1 } }  -> { 'fullDocument.stars': { '$not': { $gt: 1 } } }
+ */
+function reverseFilter<T>(filter: FilterQuery<T>) {
+  return Object.keys(filter).map(key => {
+    const value = filter[key];
+    return {
+      [`fullDocument.${key}`]: typeof value == "object" ? {$not: value} : {$ne: value}
+    };
+  });
+}
+
+function getSortedKeys(sort: {[k: string]: -1 | 1}) {
+  return Object.keys(sort);
+}
+
+function doesTheChangeAffectTheSortedCursor(
+  sortedKeys: string[],
+  change: {operationType: string; updateDescription: any}
+): boolean {
+  // If the change occurred because of an update operation, we can check if the change affects our cursor by
+  // comparing the changed keys and our sorted keys
+  if (change.operationType == "update") {
+    // TODO: Check this for subdocument updates (optimization)
+    // This check can be optimized for subdocument updates.
+    // Right now it only checks the root properties.
+    const changedKeys: string[] = (change.updateDescription.removedFields || []).concat(
+      Object.keys(change.updateDescription.updatedFields || {})
+    );
+    return changedKeys.some(k => sortedKeys.indexOf(k) > -1);
+  }
+
+  return true;
+}
+
+function fetchSortedIdsOfTheCursor<T>(
```

Comment:
@thesayyn Do you think we might need more if checks in this function?

Reply:
For now, I don't think that we'll need more "if statements" in this function.